### PR TITLE
gem5-resources: ARM Ubuntu 24.04 disk image

### PR DESCRIPTION
**Pull request by ivanaamit**
- [x] Tested 
- [x] Uploaded to the cloud bucket
- [ ] Added to gem5 resources website.

### CODE REVIEW

File: src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        ["-cpu", "host"],
+                        ["-enable-kvm"],
+                        ["-machine", "virt"],
+                        ["-machine", "gic-version=3"],
+                        ["-vga", "virtio"],
+                        ["-device","virtio-gpu-pci"],
+                        ["-device", "qemu-xhci"],
+                        ["-device","usb-kbd"],
+
+                      ]
```
Comment (powerjg) [inline src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:57]: This assumes you're running on arm hardware. Is this the same as the 22.04 image? @Harshil2107 

We need to at a minimum document, but it would be better to provide an automatic approach so that you can build this image on an Arm machine or x86 machine.

File: src/arm-ubuntu/files/gem5_init.sh [hunk 2]
```diff
@@ -19,7 +19,6 @@ fi
 if [[ $no_systemd == false ]]; then
     # gem5-bridge exit signifying that systemd will be booted
     printf "Starting systemd...\n"
-    gem5-bridge exit
```
Comment (powerjg) [inline src/arm-ubuntu/files/gem5_init.sh:22]: Why did you remove this line?

File: src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl (hunk 1 above)
Comment (Harshil2107) [reply src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:57]: yes, This is the same as the 22.04 image. Should I add an equivalent qemu command to documentation? Packer does give the qemu command it uses when we run the build with `PACKER_LOG=INFO`, and I have tried using the same args and I can boot the image in qemu.

Comment (Harshil2107) [reply src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:57]: I can figure out the args needed to build on an x86 machine and we can add it as a comment or as another packer file to build without kvm?

Comment (powerjg) [reply src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:57]: Maybe we should have two packerfiles (or if there's a way to share things between files, even better!) one for kvm and one for "other" host.

File: src/arm-ubuntu/files/gem5_init.sh (hunk 2 above)
Comment (ivanaamit) [reply src/arm-ubuntu/files/gem5_init.sh:22]: In Harshil's updated README, there is no exit event at that point.

File: src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,95 @@
+packer {
```
Comment (powerjg) [inline src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:1]: I don't know if this is in other files or not, but if it's possible we should have a copyright block.

File: src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl (hunk 1 above)
Comment (powerjg) [reply src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:57]: So this is getting solved in another PR?

File: src/arm-ubuntu/files/gem5_init.sh (hunk 2 above)
Comment (powerjg) [reply src/arm-ubuntu/files/gem5_init.sh:22]: @Harshil2107 can you confirm that all of the other init scripts no longer have this exit?

File: src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl (hunk 3 above)
Comment (Harshil2107) [reply src/arm-ubuntu/arm-ubuntu-24-04.pkr.hcl:1]: I will look into it, if I can add comments (should be able to), then I will add the copyright statement in the refactor PR, as x86 packer files also doesn't have a copyright statement.

**Comment by Harshil2107** [comment]
For the other 2 comments:

- yes, I will add the qemu args for no kvm in the refactor PR when I merge the arm packer files 
- yes, this is an extra exit event that we had decided we don't want, none of the other disk images have this exit event.

**Review by powerjg — APPROVED** [review]

**Comment by ncde** [comment]
Hello, I try to follow the README to create the arm-ubuntu image myself to install custom sw in qemu to use in gem5 later.
I could not find in the repository for example arm-ubuntu.json file, where is this located?
Also there are some unused files in gem5-resources-stable/src/arm/ubuntu/files such as after_boot.sh, when is this used?

do you have an updated README file for the instructions?

Sorry I write comment to your commit but I could not find the place to create an issue probably it is disabled.
Best,

**Comment by Harshil2107** [comment]
Hi,

The source to build the arm image is at https://github.com/gem5/gem5-resources/tree/stable/src/arm-ubuntu

We use `.pck.hcl` files instead of  `.json`. The update readme is currently at a PR this is the link to the documentation to build this image: https://github.com/gem5/gem5-resources/blob/8b4955fa28b2c28bee4fb46e182137ca708d5d30/src/ubuntu-generic-diskimages/BUILDING.md. Sorry for the confusion, if you have any questions, I am happy to answer.

We use the `after_boot.sh` when we first login to a shell, so that we can run the readfile that is passed in gem5.

**Comment by ncde** [comment]
thank you very much I will follow the building document
